Backpack/Dialog/Classes/BPKDialogView.m

```diff
@@ -0,0 +1,221 @@
+                                              [self.descriptionLabel.topAnchor constraintEqualToAnchor:self.titleLabel.bottomAnchor constant:BPKSpacingMd],
+                                              [self.descriptionLabel.leadingAnchor constraintEqualToAnchor:self.descriptionLabel.superview.layoutMarginsGuide.leadingAnchor],
+                                              [self.descriptionLabel.superview.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.descriptionLabel.trailingAnchor],
+                                              
+                                              [self.buttonStackView.topAnchor constraintEqualToAnchor:self.descriptionLabel.bottomAnchor constant:BPKSpacingLg],
+                                              [self.buttonStackView.leadingAnchor constraintEqualToAnchor:self.buttonStackView.superview.layoutMarginsGuide.leadingAnchor constant:(BPKSpacingXl - BPKSpacingLg)],
+                                              [self.buttonStackView.superview.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.buttonStackView.trailingAnchor constant:(BPKSpacingXl - BPKSpacingLg)],
+                                              [self.buttonStackView.superview.layoutMarginsGuide.bottomAnchor constraintEqualToAnchor:self.buttonStackView.bottomAnchor]
+                                              ]];
+}
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+
+    if (self.hasShadow) {
+        [[BPKShadow shadowLg] applyToLayer:self.layer];
+    } else {
+        [self.layer setShadowOpacity:0];
+    }
+}
+
+-(void)setHasShadow:(BOOL)hasShadow {
+    _hasShadow = hasShadow;
+    [self setNeedsLayout];
+}
+
+- (void)buttonTapped:(BPKButton *)button {
+    BPKDialogButtonAction *action = [self actionForButton:button];
+    if (action) {
+        [self.delegate didInvokeButtonAction:action];
+    }
+}
+
+- (BPKDialogButtonAction * _Nullable)actionForButton:(BPKButton *)button {
+    for (BPKActionButtonPair *pair in self.registeredActions) {
+        if (pair.button == button) {
+            return pair.action;
+        }
+    }
+
+    return nil;
+}
+
+#pragma mark - Public
+
+
+#pragma mark - Property overrides
+
+- (void)setIconBackgroundColor:(UIColor *_Nullable)iconBackgroundColor {
+    self.iconView.iconBackgroundColor = iconBackgroundColor;
+}
+
+- (UIColor *_Nullable)iconBackgroundColor {
+    return self.iconView.iconBackgroundColor;
+}
+
+- (void)setIconImage:(UIImage *_Nullable)iconImage {
+    self.iconView.iconImage = iconImage;
+}
+
+- (UIImage *_Nullable)iconImage {
+    return self.iconView.iconImage;
+}
+
+- (void)setTitle:(NSString *_Nullable)title {
+    self.titleLabel.text = title;
+}
+
+- (NSString *_Nullable)title {
+    return self.titleLabel.text;
+}
+
+
+- (void)setDescription:(NSString *_Nullable)description {
+    self.descriptionLabel.text = description;
+}
+
+- (NSString *_Nullable)description {
+    return self.descriptionLabel.text;
```

Comment:
overriding description will stop having the memory address when printing it

Reply:
Really cood catch 👍